datafusion/common/src/config.rs

```diff
@@ -349,6 +349,11 @@ config_namespace! {
         /// metadata memory consumption
         pub batch_size: usize, default = 8192
 
+        /// Minimum batch size before DataFusion will attempt to
+        /// split oversized record batches coming from sources.
+        /// Set to 0 to disable splitting entirely.
+        pub batch_split_threshold: usize, default = 8192
```

Comment:
I can't think of any usecase where this should be something different than the existing `batch_size`

Thus I suggest removing this new config and always splitting using batch_size

https://github.com/apache/datafusion/blob/d66d6b9eb83a4643bc8d1ee9c2da7f0fa644c0cf/datafusion/common/src/config.rs#L350-L349